src/Language/PureScript/Errors.hs

```diff
@@ -858,14 +854,21 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath fileCon
             , line "has escaped its scope, appearing in the type"
             , markCodeBox $ indent $ prettyType ty
             ]
-    renderSimpleErrorMessage (TypesDoNotUnify u1 u2)
+    renderSimpleErrorMessage (TypesDoNotUnify isOrdered u1 u2)
       = let (row1Box, row2Box) = printRows u1 u2
 
-        in paras [ line "Could not match type"
-                 , row1Box
-                 , line "with type"
-                 , row2Box
-                 ]
+        in if isOrdered then
+          paras [ line "The type"
+                , row1Box
+                , line "does not match the expected type"
```

Comment:
Fixed in latest commit.

Reply:
There's a good case for using the word ‘actual’ here—GHC does it, I used it when talking about the change informally, it pairs well with ‘expected’—but I feel a little odd about using it in the error message. When you try to pass an `Int` into a function of type `String -> Boolean`, both `Int` and `String` are ‘actual’ types—neither type is more real than the other.

I guess you could make a similar argument that the compiler can ‘expect’ the user to provide a term of type `String` to that function, or it could ‘expect’ the user to provide a function term of type `Int -> a`, but my inclination is to apply a little anthropomorphism and say that the *function* is doing the ‘expecting’ and expects a `String` and not an `Int`.

When checking an annotated type `(3 :: String)`, expected/actual makes more sense. But maybe there's a term that could replace ‘actual’ for both cases?

Reply:
Other renderings I've seen are
> Expected X but got Y

So, the error message could be updated to say

```
Expected type

   X
   
but got type

   Y
```